### DISCUSSION TITLE
Add extraction utility for Flash Develop

### DISCUSSION THREAD

**Discussion by marchbold** [airsdk/apm#106] [Ideas]
FlashDevelop does not automatically extract extensions into a directory for debugging, so you have to provide not only the ANE but the extracted ANE (which directory name has to match the ANE)

We could add some additional functionality that automatically extracts the ane's into a specified directory for you

Something like 

```
apm ide flashdevelop configure 
```

- Is this still a requirement of the latest version of flash develop? 
- What would the required options be? Is an output directory required or is this consistent ? 

I have been considering adding in some generators for specific IDEs, mainly around project config, but this could be a good starting point.

**Comment by Ender22** [comment]
Is this still a requirement of the latest version of flash develop?

- Yes, flash develop hasn't been updated in forever so this is needed for most recent version

What would the required options be? Is an output directory required or is this consistent?

- It looks like the ane's can be extracted to any directory. The developer just needs to go to that folder inside of FlashDevelop and right-click each extracted folder and choose the 'add to library' option. I think it would be safest to let the developer choose the path to this output folder once to fit different project structures but it likely wouldn't change after that. 

I guess the extraction function would also need to handle the 'replacing/overwriting' of the folders so that they don't need to be manually deleted before updates are made too.
